### PR TITLE
feature: add integration with firebase for topics/subtopics

### PR DESCRIPTION
**Pull request by nicmalegre**
### Pull request type

- [ ] Bugfix
- [X] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

---

### Description

- Get topics and subtopics from firebase
- Create hooks to add that logic
- FYI: In next PR the firebase queries will be refactor 

---

#### Jira board reference

- [Explore all topics](https://rootstrap.atlassian.net/browse/TH-16?atlOrigin=eyJpIjoiMjFmZjJlNzU4NjQxNDU4MzgyZTVhZDBiMzJmOGExZTAiLCJwIjoiaiJ9)

---

#### Notes:

-

---

#### Tasks:

- [X] Tested on IOS
- [ ] Tested on Android
- [ ] Tested on a small device
- [ ] Tested on a real device
- [ ] Tested all flows related with this PR changes
- [ ] Tested accessibility
- [ ] Added tests

---

#### Preview

**Firebase:**
Topics:
Screenshot 2023-03-17 at 15 48 09

Subtopics:
Screenshot 2023-03-17 at 15 48 25

**IOS:**

https://user-images.githubusercontent.com/40902023/225993683-452f257b-5e4b-453f-9bdb-95566efc3e92.mp4

### CODE REVIEW

**Review by jpirazusta — APPROVED** [review]
🙌

File: src/features/home/hooks/useGetTopics.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import firestore from '@react-native-firebase/firestore';
+
+import { getErrorMessage } from 'utils/error';
+
+import { Collections } from 'constants/firebase';
+
+import { Topics } from 'features/home/components/types';
+
+export const useGetTopics = () => {
+  const [topics, setTopics] = useState<Topics | []>([]);
```
Comment (ManuViola77) [inline src/features/home/hooks/useGetTopics.ts:12]: do you need the `| []`? isn't the empty array included as a possibility in the array type?

File: src/features/home/screen/index.tsx [hunk 2]
```diff
@@ -59,16 +29,16 @@ const HomeScreen: FunctionComponent<HomePropTypes> = ({ navigation: { navigate }
...
     <SafeAreaView sx={styles.container}>
       <ScrollView style={styles.scrollView}>
         <Text accessibilityRole="text" variant="h4" sx={styles.title}>
           {strings.HOME.title}
         </Text>
 
-        {TOPICS.map(topic => (
-          <Topic key={topic.name} topic={topic} />
-        ))}
+        {!!topics.length && topics.map(topic => <Topic key={topic.name} topic={topic} />)}
```
Comment (ManuViola77) [inline src/features/home/screen/index.tsx:41]:
```suggestion
        {!!topics.length && topics.map(topic => <Topic key={topic.id} topic={topic} />)}
```

I think you can use the id

**Review by ManuViola77 — APPROVED** [review]
Great work 👏🏻

File: src/features/home/screen/index.tsx (hunk 2 above)
Comment (nicmalegre) [reply src/features/home/screen/index.tsx:41]: Yeah, I forgot to update. Thanks

**Pull request merged**